lib/yard/parser/ruby/ruby_parser.rb

```diff
@@ -144,7 +144,7 @@ def file_encoding
         PARSER_EVENT_TABLE.each do |event, arity|
           node_class = AstNode.node_class_for(event)
 
-          if /_new\z/ =~ event.to_s && arity == 0
+          if arity == 0 && /_new\z/ =~ event.to_s
```

Comment:
Could the regex check be replaced with `event.to_s.end_with?('_new')`

Reply:
arity is better since ends_with breaks prior Ruby compatibility.

Reply:
What’s the minimum supported ruby version? The [gemspec](https://github.com/lsegal/yard/blob/main/yard.gemspec#L24) says it depends on webrick 1.7.0, which [requires](https://rubygems.org/gems/webrick/versions/1.7.0) ruby >= 2.3.0, which [supports](https://ruby-doc.org/core-2.3.0/String.html) `String#end_with?`

Reply:
Good to know that end_with is available, but a little moot since arity is already here :)